app/policies/comment_policy.rb

```diff
@@ -0,0 +1,27 @@
+# app/policies/comment_policy.rb
+
+class CommentPolicy < ApplicationPolicy
+  attr_reader :user, :comment
+
+  def initialize(user, comment)
+    @user = user
+    @comment = comment
+  end
+
+  def show?
+    user == comment.author || PhotoPolicy.new(user, comment.photo).show?
```

Comment:
Can we simplify the second part with just `policy(comment.photo).show?`? Maybe debatable if we even need this logic here